app/presenters/wpcc/message_presenter.rb

```diff
@@ -1,5 +1,18 @@
 module Wpcc
   class MessagePresenter < Presenter
+    def self.opt_in_thresholds
+      config = {}
+
+      ::Wpcc::SalaryMessage::OPT_IN_THRESHOLDS.each do |threshold, frequencies|
+        frequencies.each do |frequency, frequency_threshold_value|
+          config[frequency] ||= {}
+          config[frequency][threshold] = frequency_threshold_value
+        end
+      end
+
+      config
+    end
+
```

Comment:
Probably worth documenting why it is necessary to convert the opt_in_thresholds hash.

Reply:
Done in the commit message

Reply:
Nice documentation in the commit message. There are a few typos in that message and it is somewhat hard to follow. Is it worth re-wording e.g.

```
The Dough convention for loading components, e.g data-dough-component=SalaryConditions, is:
the html attribute called 'data-dough' PLUS
the name of the component PLUS 
'config' 
(e.g data-dough-salary-conditions-config).
Dough then parses the object and stores the data in the config variable, which is the second
argument of the initialise function.

Because the javascript and backend format the salary_threshold data differently, it is necessary to convert the backend format to the format required by the js.

Prior to this change, the JS 'optInTriggers' variable was on local
scope of the salary conditions file. Now it is an attribute of
the object salary conditions which necessitates using 'this' to ensure the correct scope.
```